### ISSUE TITLE
setAudioSource fails when initialIndex is not within range of the previous audio source.

### ISSUE THREAD

**Issue by colehorvitz** [ryanheise/just_audio#644]
**Which API doesn't behave as documented, and how does it misbehave?**
`setAudioSource` fails when an initial index is provided which is outside the range of the previously loaded audio source. To be clear, this only occurs when `setAudioSource` is called multiple times, i.e. when an audio source has previously been set. The method fails when `initialIndex` is not within the previous `AudioSource`'s range, even though it is within the new `AudioSource`'s range. So far, I have only tested this on iOS.

Please let me know if I am using this API incorrectly.

**Minimal reproduction project**
https://github.com/colehorvitz/just_audio/tree/master/just_audio/example

I have modified `example_playlist.dart` to demonstrate this behavior. I have created two playlists, `_playlist1`, which contains one item, and `_playlist2`, which contains three items. `_playlist1` is loaded by default when the app opens with an `initialIndex` of 0. I have furthermore modified the `FloatingActionButton`'s `onPressed` callback to demonstrate the error. When this button is pressed, it executes `await _player.setAudioSource(_playlist2, initialIndex: 1)`. However, since index 1 is outside of the range of `playlist1` (which only contains 1 item and thus only contains index 0 in its range), an error occurs, even though index 1 is within the range of `_playlist2`, which contains three items.  ​

This error does not occur when executing the code `await _player.setAudioSource(_playlist2, initialIndex: 0)`. 

**To Reproduce (i.e. user steps, not code)**
1. Run the example project using `flutter run -t lib/example_playlist.dart`
2. Wait for the first audio source to load with its singular item
3. Press the floating action button, which will set the audio source to the second playlist with an initial index of 1

**Error messages**
When `setAudioSource` is called:
```
Unhandled Exception: (9999999) Error in handleMethodCall
#0      AudioPlayer._load (package:just_audio/just_audio.dart:783:11)
<asynchronous suspension>
#1      AudioPlayer.load (package:just_audio/just_audio.dart:708:14)
<asynchronous suspension>
#2      AudioPlayer.setAudioSource (package:just_audio/just_audio.dart:683:18)
<asynchronous suspension>
#3      _MyAppState.build.<anonymous closure> (package:just_audio_example/example_playlist.dart:289:13)
<asynchronous suspension>
[VERBOSE-2:ui_dart_state.cc(209)] Unhandled Exception: PlatformException(error, Error in handleMethodCall, null, null)
#0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:607:7)
#1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:167:18)
<asynchronous suspension>
#2      MethodChannelAudioPlayer.play (package:just_audio_platform_interface/method_channel_just_audio.dart:56:34)
<asynchronous suspension>
#3      AudioPlayer._sendPlayRequest (package:just_audio/just_audio.dart:890:7)
<asynchronous suspension>
```

When the player is subsequently played after receiving this error:
```
*** Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[__NSArrayM objectAtIndexedSubscript:]: index 1 beyond bounds [0 .. 0]'
*** First throw call stack:
(0x18395edc0 0x1984bb7a8 0x1839d15a0 0x183849568 0x1020fc53c 0x1020fe1a8 0x1020fffac 0x184cc5e1c 0x184cc8394 0x184cc55c8 0x18b3d38d4 0x183549a84 0x18354b81c 0x183559c70 0x1838d8a2c 0x1838d2904 0x1838d19f4 0x19afa7734 0x18635075c 0x186355fcc 0x1020f230c 0x18358dcf8)
libc++abi: terminating with uncaught exception of type NSException
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
    frame #0: 0x00000001b1b9f334 libsystem_kernel.dylib`__pthread_kill + 8
libsystem_kernel.dylib`__pthread_kill:
->  0x1b1b9f334 <+8>:  b.lo   0x1b1b9f354               ; <+40>
    0x1b1b9f338 <+12>: pacibsp
    0x1b1b9f33c <+16>: stp    x29, x30, [sp, #-0x10]!
    0x1b1b9f340 <+20>: mov    x29, sp
Target 0: (Runner) stopped.
```

**Expected behavior**
I would expect the new audio source to be loaded and seek to an initial index if it is within range of the new audio source.

**Smartphone (please complete the following information):**
 - Device: iPhone10
 - OS:  iOS14.8.1

**Flutter SDK version**
```
[✓] Flutter (Channel stable, 2.8.1, on macOS 12.0.1 21A559 darwin-x64, locale en-US)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.63.2)
[✓] Connected device (3 available)
```

**Comment by ryanheise** [comment]
Thanks for finding and reporting this bug, I'll take a look.

**Comment by colehorvitz** [comment]
Thanks Ryan. In the meantime, 
```
await _player.setAudioSource(_playlist);
await _player.seek(Duration.zero, initialIndex);
```
seems to get the job done.

**Comment by ryanheise** [comment]
I have created a bug fix on the branch `fix/ios_initial_index`. Let me know how it goes and I'll make a new release if it works for you.

**Comment by ryanheise** [comment]
Have you had a chance to test this? I'd like to push out a release but would appreciate confirmation from you whether it actually worked in your app.

**Comment by colehorvitz** [comment]
Just got the chance to try it. Seems to be working now!

**Comment by ryanheise** [comment]
Great to hear it worked for you. I have now published this fix in release 0.9.19.
